src/clip.js

```diff
@@ -41,13 +41,17 @@ function clip(features, scale, k1, k2, axis, intersect, minAll, maxAll) {
         if (slices.length) {
             // if a feature got clipped, it will likely get clipped on the next zoom level as well,
             // so there's no need to recalculate bboxes
-            clipped.push({
+            var clippedFeature = {
                 geometry: slices,
                 type: type,
                 tags: features[i].tags || null,
                 min: feature.min,
                 max: feature.max
-            });
+            };
+            if (typeof feature.id !== 'undefined') {
+                clippedFeature.id = feature.id;
+            }
```

Comment:
@jfirebaugh it's important for properties, but I don't think that's the case for feature ids. Also, if we add the `id` property after defining the object without it, like in the code above, V8 has to switch a hidden class for the object every time, slowing things down, so it's a performance antipattern. It may not be significant but we should measure and make a call.